client/extensions/woocommerce/app/dashboard/index.js

```diff
@@ -20,55 +23,68 @@ class Dashboard extends Component {
 		className: PropTypes.string,
 	};
 
-	onStoreSetupFinished = () => {
-		// TODO - save that setup has been finished to the store's state on WPCOM
-		// TODO - is there a way to set an option on the store site?
+	componentDidMount = () => {
+		const { selectedSite } = this.props;
+
+		if ( selectedSite && selectedSite.ID ) {
+			this.props.fetchSetupChoices( selectedSite.ID );
+		}
 	}
 
-	renderStoreSetup = () => {
+	componentWillReceiveProps = ( newProps ) => {
 		const { selectedSite } = this.props;
-		return (
-			<Setup
-				onFinished={ this.onStoreSetupFinished }
-				site={ selectedSite }
-			/>
-		);
-	}
 
-	renderStoreManagement = () => {
-		const { translate } = this.props;
+		const newSiteID = newProps.selectedSite && newProps.selectedSite.ID || null;
```

Comment:
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/fc929a9089c59738059bec4d882378d93712696e